sdk/textanalytics/ai-text-analytics/src/generated/generatedClient.ts

```diff
@@ -1,211 +1,154 @@
 /*
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License. See License.txt in the project root for
- * license information.
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
  *
  * Code generated by Microsoft (R) AutoRest Code Generator.
- * Changes may cause incorrect behavior and will be lost if the code is
- * regenerated.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
  */
 
 import * as coreHttp from "@azure/core-http";
+import * as Parameters from "./models/parameters";
 import * as Models from "./models";
 import * as Mappers from "./models/mappers";
-import * as Parameters from "./models/parameters";
 import { GeneratedClientContext } from "./generatedClientContext";
+import {
+  GeneratedClientOptionalParams,
+  MultiLanguageBatchInput,
+  GeneratedClientEntitiesRecognitionGeneralOptionalParams,
+  GeneratedClientEntitiesRecognitionGeneralResponse,
+  GeneratedClientEntitiesLinkingOptionalParams,
+  GeneratedClientEntitiesLinkingResponse,
+  GeneratedClientKeyPhrasesOptionalParams,
+  GeneratedClientKeyPhrasesResponse,
+  LanguageBatchInput,
+  GeneratedClientLanguagesOptionalParams,
+  GeneratedClientLanguagesResponse,
+  GeneratedClientSentimentOptionalParams,
+  GeneratedClientSentimentResponse
+} from "./models";
 
 class GeneratedClient extends GeneratedClientContext {
   /**
    * Initializes a new instance of the GeneratedClient class.
    * @param endpoint Supported Cognitive Services endpoints (protocol and hostname, for example:
-   * https://westus.api.cognitive.microsoft.com).
-   * @param [options] The parameter options
+   *                 https://westus.api.cognitive.microsoft.com).
+   * @param options The parameter options
    */
-  constructor(endpoint: string, options?: coreHttp.ServiceClientOptions) {
+  constructor(endpoint: string, options?: GeneratedClientOptionalParams) {
```

Comment:
@joheredi, @sarangan12,

Why is it that we have the `GeneratedClient` prefix in many places? For packages that will have a convenience layer over the generated layer this is not a problem, but those that use the code generator output directly, such naming would not be user friendly right?

Reply:
`GeneratedClient` is added in this library as a title override. The generator would compose the name based on the title given in the Swagger, so there won't be `Generated` prefixed added by the generator

This is what this library has in the swagger/readme.md

```yaml
package-name: "@azure/ai-text-analytics"
title: GeneratedClient
...
```

Reply:
Ah, interesting. Should we go with just `Generated` instead? The fact that all this is for a "client" is more obvious isnt it?

Reply:
We use GeneratedClient because our internal docs systems can't handle packages that have two types that are the same name exported from two different files. If we used just `Generated` as the prefix with the current code gen, we would get a codegen client that is just named `{Prefix}` (i.e. `new Generated(...)`), but if we could have the client be output as `{Prefix}Client`, then that would make this a little less verbose for us.

Reply:
This is good feedback thanks @willmtemple. I have filed Azure/autorest.typescript/issues/703 to look into adding Client as a suffix so that when defining the name of the library adding the Client word is not needed.